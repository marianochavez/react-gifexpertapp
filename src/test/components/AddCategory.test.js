import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { AddCategory } from "../../components/AddCategory";



describe("Test AddCategory", () => {

    const setCategory = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategory}/>);
    
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategory}/>);
    });
    
    test('should show AddCategory correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should change the text box', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';

        input.simulate('change',{ target: {value} });

        expect(wrapper.find('p').text().trim()).toBe(value);

    })

    test('should not add a category if it is empty', () => {

        wrapper.find('form').simulate('submit',{preventDefault(){}});

        expect(setCategory).not.toHaveBeenCalled();
    })
    
    test('should call setCategories and clean the text box', () => {
        const value = 'Pokemon';
        // const input = wrapper.find('input');
        wrapper.find('input').simulate('change', {target:{value}});
        wrapper.find('form').simulate('submit',{preventDefault(){}});

        expect(setCategory).toHaveBeenCalled();
        expect(wrapper.find('input').props().value).toBe('');
    })
    

});