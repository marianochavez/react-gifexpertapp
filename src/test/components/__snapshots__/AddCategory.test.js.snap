// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddCategory should show AddCategory correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
